modules/frontend/tracebyidsharding.go

```diff
@@ -65,7 +65,8 @@ func (s shardQuery) RoundTrip(r *http.Request) (*http.Response, error) {
 
 	var overallError error
 	var totalFailedBlocks uint32
-	overallTrace := &tempopb.Trace{}
+	combiner := trace.NewCombiner()
+	combiner.Consume(&tempopb.Trace{}) // The query path returns a non-nil result even if no inputs (which is different than other paths which return nil for no inputs)
```

Comment:
This is another fiddly bit.  Some paths expect nil if no inputs, but this read path defaults to an empty trace.  I think it might be cleaner to make this an option in `NewCombiner` to control the default result:  either nil or empty trace.

Reply:
Downstream from this is the logic to determine 2xx/4xx/5xx from the responses, wasn't clear if nil could even be interpreted differently.  Therefore agree this seems better than introducing a change there, at least for now and we can always come back to it if not.